integration_tests/support/sharedTests.ts

```diff
@@ -143,6 +187,41 @@ const sharedTests = {
       programmeHistoryPage.shouldContainNoHistoryParagraph()
     },
 
+    showsOffenceHistoryPage: (role: ApplicationRole): void => {
+      sharedTests.referrals.beforeEach(role)
+      cy.task('stubOffenderBooking', offenderBooking)
+      cy.task('stubOffenceCode', offenceCodeOne)
+      cy.task('stubOffenceCode', offenceCodeTwo)
+
+      const path = pathsByRole(role).show.offenceHistory({ referralId: referral.id })
+      cy.visit(path)
+
+      const offenceHistoryPage = Page.verifyOnPage(OffenceHistoryPage, {
+        course,
+        person,
+      })
+      offenceHistoryPage.shouldHavePersonDetails(person)
+      offenceHistoryPage.shouldContainNavigation(path)
+      offenceHistoryPage.shouldContainBackLink('#')
+      offenceHistoryPage.shouldContainSubmissionSummaryList(referral)
+      offenceHistoryPage.shouldContainSubmittedReferralSideNavigation(path, referral.id)
+      offenceHistoryPage.shouldContainImportedFromText()
+      offenceHistoryPage.shouldContainIndexOffenceSummaryCard({
+        code: offenceCodeOne.code,
+        date: offenderBooking.offenceHistory?.[0].offenceDate,
+        description: offenceCodeOne.description,
+        mostSerious: offenderBooking.offenceHistory?.[0].mostSerious,
+        statuteCodeDescription: offenceCodeOne.statuteCode.description,
+      })
+      offenceHistoryPage.shouldContainAdditionalOffenceSummaryCard({
+        code: offenceCodeTwo.code,
+        date: offenderBooking.offenceHistory?.[1].offenceDate,
+        description: offenceCodeTwo.description,
+        mostSerious: offenderBooking.offenceHistory?.[1].mostSerious,
+        statuteCodeDescription: offenceCodeTwo.statuteCode.description,
+      })
```

Comment:
Could we pass the booking and codes to the page, set them as properties, and then move this detail over there?